lib/internal/repl.js

```diff
@@ -165,3 +156,60 @@ function _replHistoryMessage() {
   this._historyPrev = Interface.prototype._historyPrev;
   return this._historyPrev();
 }
+
+
+function getTMPFile(ready) {
+  var tmpPath = os.tmpdir();
+  if (!tmpPath) {
+    return ready(new Error('no tmpdir available'));
+  }
+  tmpPath = path.join(tmpPath, process.pid + '-' + 'repl.tmp');
+  fs.open(tmpPath, 'w', 0o600, function(err, fd) {
+    if (err) {
+      return ready(err);
+    }
+    return ready(null, {
+      fd: fd,
+      path: tmpPath
+    });
+  });
+}
+
+// write data, sync the fd, close it, overwrite the target
+// with a rename, open a new tmpfile
+function writeAndSwapTMP(data, tmpInfo, target, ready) {
+  debug('writing tmp file... %s', tmpInfo.path, data);
+  return fs.write(tmpInfo.fd, data, 0, 'utf8', onwritten);
+
+  function onwritten(err) {
+    if (err) return ready(err);
+    debug('syncing... %s', tmpInfo.path);
+    fs.fsync(tmpInfo.fd, onsync);
+  }
+
+  function onsync(err) {
+    if (err) return ready(err);
+    debug('closing... %s', tmpInfo.path);
+    fs.close(tmpInfo.fd, onclose);
+  }
+
+  function onclose(err) {
+    if (err) return ready(err);
+    debug('rename... %s -> %s', tmpInfo.path, target);
+    fs.rename(tmpInfo.path, target, onrename);
```

Comment:
I could swear I'd replied... a race window between creating and renaming the temporary file.  In theory, something like a cron job can delete the file in that time window.  There is also possibly an issue with multiple iojs processes terminating at the same (e.g. because the login session is terminated.)